gradebookng/tool/src/java/org/sakaiproject/gradebookng/business/GradebookNgBusinessService.java

```diff
@@ -266,16 +266,45 @@ private Gradebook getGradebook(String siteId) {
 	}
 	
 	/**
-	 * Get a list of assignments in the gradebook in the specified site that the current user is allowed to access, sorted by sort order
-	 * 
+	 * Get a list of assignments in the gradebook in the specified site that the current user is allowed to access, sorted by default sort order
+	 *
 	 * @param siteId the siteId
 	 * @return a list of assignments or null if no gradebook
 	 */
 	public List<Assignment> getGradebookAssignments(String siteId) {
+		return getGradebookAssignments(siteId, false);
+	}
+
+	/**
+	 * Get a list of assignments in the gradebook in the specified site that the current user is allowed to access, sorted by sort order or by categorised sort order
+	 *
+	 * @param sortByCategory whether to pre-sort the assignments based on their categorized sort order
+	 * @return a list of assignments or null if no gradebook
+	 */
+	public List<Assignment> getGradebookAssignments(boolean sortByCategory) {
```

Comment:
This is fine for now, the GradebookUiSettings was meant to be a model for the front end. If we need more sort params we could look at moving it around then I reckon.